### PR TITLE
Concat date with start & end times

### PR DESCRIPTION
**Pull request by ilikesymmetry**
Concat date with start and end times to distinguish when operating hours extend into the next day.  Standardization of hours is as follows

isodate -> YYYY-MM-DD
time      ->  hh:mm<am or pm>

Now new with concat:
start_time -> isodate:time
end_time  -> isodate:time

### CODE REVIEW

File: data.py [hunk 1]
```diff
@@ -236,17 +238,31 @@ def parse_static_op_hours(hours_list, eatery_id):
...
+  [end_hour, end_minute] = end_time.split(':')
+  start_hour = '0' + start_hour if len(start_hour) < 2 else start_hour
+  end_hour = '0' + end_hour if len(end_hour) < 2 else end_hour
+  end_date = start_date
+  if (int(end_hour) < int(start_hour) or end_hour == '12') and end_minute[-2:] == 'am':
+    [year, month, day] = start_date.split('-')
+    next_day = date(int(year), int(month), int(day)) + timedelta(days=1)
+    end_date = next_day.isoformat()
+  new_start = start_date + ':' + start_hour + ':' + start_minute
+  new_end = end_date + ':' + end_hour + ':' + end_minute
```
Comment (young-k) [inline data.py]: format string for this line and the above line as well

File: data.py [hunk 2]
```diff
@@ -236,17 +238,31 @@ def parse_static_op_hours(hours_list, eatery_id):
...
     new_operating_hours.append(new_operating_hour)
   operating_hours[eatery_id] = new_operating_hours
   return new_operating_hours
 
+def format_time(start_time, end_time, start_date):
+  [start_hour, start_minute] = start_time.split(':')
+  [end_hour, end_minute] = end_time.split(':')
+  start_hour = '0' + start_hour if len(start_hour) < 2 else start_hour
+  end_hour = '0' + end_hour if len(end_hour) < 2 else end_hour
+  end_date = start_date
```
Comment (young-k) [inline data.py:258]: what's the point of this?

File: data.py [hunk 3]
```diff
@@ -236,17 +238,31 @@ def parse_static_op_hours(hours_list, eatery_id):
...
     )
     new_operating_hours.append(new_operating_hour)
   operating_hours[eatery_id] = new_operating_hours
   return new_operating_hours
 
+def format_time(start_time, end_time, start_date):
+  [start_hour, start_minute] = start_time.split(':')
+  [end_hour, end_minute] = end_time.split(':')
+  start_hour = '0' + start_hour if len(start_hour) < 2 else start_hour
+  end_hour = '0' + end_hour if len(end_hour) < 2 else end_hour
```
Comment (young-k) [inline data.py]: replace with:
`end_hour = end_hour.rjust(2, "0")`

@kq-li @alanna-zhou @meganle lmk is this nifty or is this nifty

File: data.py [hunk 4]
```diff
@@ -236,17 +238,31 @@ def parse_static_op_hours(hours_list, eatery_id):
...
+        events=parse_events(new_events, eatery_id, new_date.isoformat()),
         status='EVENTS'
     )
     new_operating_hours.append(new_operating_hour)
   operating_hours[eatery_id] = new_operating_hours
   return new_operating_hours
 
+def format_time(start_time, end_time, start_date):
+  [start_hour, start_minute] = start_time.split(':')
+  [end_hour, end_minute] = end_time.split(':')
```
Comment (young-k) [inline data.py]: you can omit the `[]`, so `end_hour, end_minute = end_time.split(':')`

File: data.py [hunk 5]
```diff
@@ -236,17 +238,31 @@ def parse_static_op_hours(hours_list, eatery_id):
...
+  start_hour = '0' + start_hour if len(start_hour) < 2 else start_hour
+  end_hour = '0' + end_hour if len(end_hour) < 2 else end_hour
+  end_date = start_date
+  if (int(end_hour) < int(start_hour) or end_hour == '12') and end_minute[-2:] == 'am':
+    [year, month, day] = start_date.split('-')
```
Comment (young-k) [inline data.py]: same here as well, the `[]` can be omitted

File: data.py [hunk 6]
```diff
@@ -236,17 +238,31 @@ def parse_static_op_hours(hours_list, eatery_id):
...
   operating_hours[eatery_id] = new_operating_hours
   return new_operating_hours
 
+def format_time(start_time, end_time, start_date):
+  [start_hour, start_minute] = start_time.split(':')
+  [end_hour, end_minute] = end_time.split(':')
+  start_hour = '0' + start_hour if len(start_hour) < 2 else start_hour
+  end_hour = '0' + end_hour if len(end_hour) < 2 else end_hour
+  end_date = start_date
+  if (int(end_hour) < int(start_hour) or end_hour == '12') and end_minute[-2:] == 'am':
```
Comment (young-k) [inline data.py]: `end_minute.endswith('am')`

**Review by young-k — CHANGES_REQUESTED** [review]
Some cool pythonic changes we can implement to have cleaner code

**Review by young-k — APPROVED** [review]
LGTM

File: data.py (hunk 3 above)
Comment (alanna-zhou) [reply data.py]: wau so nifty where did you learn this from

Comment (young-k) [reply data.py]: i read all of `python` documentation when i was preparing for interviews hahAA

File: data.py (hunk 1 above)
Comment (kq-li) [reply data.py]: make sure to use python 3 [f-strings](https://realpython.com/python-f-strings/)!

File: data.py (hunk 3 above)
Comment (kq-li) [reply data.py]: spending your time wisely i see

Comment (meganle) [reply data.py]: wOw so nifty

**Pull request merged**